spark/src/main/java/org/apache/zeppelin/spark/PySparkInterpreter.java

```diff
@@ -423,8 +424,15 @@ public int getProgress(InterpreterContext context) {
     }
 
     synchronized (statementFinishedNotifier) {
-      while (statementOutput == null) {
+      long startTime = System.currentTimeMillis();
+      while (statementOutput == null
+        && pythonScriptInitialized == false
+        && pythonscriptRunning) {
         try {
+          if (System.currentTimeMillis() - startTime < MAX_TIMEOUT_SEC * 1000) {
+            logger.error("pyspark completion didn't have response for 10sec.");
```

Comment:
Shall we replace hard-coded constant value here with  `MAX_TIMEOUT_SEC` ?

Also, to get pretty stacktrace in case of this error - it's better to `logger.error("...", e);`